### PR TITLE
fix: fixed responsiveness issues on mobile view

### PR DESCRIPTION
**Pull request by eemaanamir**
[](url)Ticket: [INF-1708](https://2u-internal.atlassian.net/browse/INF-1708)

Fixed responsiveness issues on mobile and tablet views. 

### Laptop View 1440px:
localhost_1995_u_edx (1)

### Laptop View 1024px:
localhost_1995_u_edx (2)

### Tablet View 768px:
localhost_1995_u_edx (3)

### Mobile View 425px:
localhost_1995_u_edx (4)

### Mobile View 375px:
localhost_1995_u_edx (5)

### Mobile View 320px: 
localhost_1995_u_edx (6)

### CODE REVIEW

**Review by AhtishamShahid — APPROVED** [review]

File: src/profile-v2/CertificateCard.jsx [hunk 1]
```diff
@@ -37,24 +40,47 @@ const CertificateCard = ({
...
+          { 'width-19625rem': !isMobileView },
+        )}
+        >
           <div className="w-100 color-black">
-            <p className="small mb-0 font-weight-normal">
+            <p className={classNames([
+              'mb-0 font-weight-normal',
+              isMobileView ? 'x-small' : 'small',
+            ])}
+            >
```
Comment (sundasnoreen12) [inline src/profile-v2/CertificateCard.jsx:54]: follow this pattern for better readability 
<p className={classNames('mb-0 font-weight-normal', {
  'x-small': isMobileView,
  'small': !isMobileView,
})}>

File: src/profile-v2/CertificateCard.jsx [hunk 2]
```diff
@@ -37,24 +40,47 @@ const CertificateCard = ({
...
                 messages['profile.certificates.types.unknown'],
               ))}
             </p>
-            <h4 className="m-0 color-black">{courseDisplayName}</h4>
-            <p className="small mb-0">
+            <p className={classNames([
+              'm-0 color-black',
+              isMobileView ? 'h5' : 'h4',
+            ])}
+            >
```
Comment (sundasnoreen12) [inline src/profile-v2/CertificateCard.jsx:65]: <p className={classNames('m-0 color-black', {
  'h5': isMobileView,
  'h4': !isMobileView,
})}>

File: src/profile-v2/CertificateCard.jsx [hunk 3]
```diff
@@ -37,24 +40,47 @@ const CertificateCard = ({
...
+              {courseDisplayName}
+            </p>
+            <p className={classNames([
+              'mb-0',
+              isMobileView ? 'x-small' : 'small',
+            ])}
+            >
```
Comment (sundasnoreen12) [inline src/profile-v2/CertificateCard.jsx:72]: <p className={classNames('mb-0', {
  'x-small': isMobileView,
  'small': !isMobileView,
})}>

File: src/profile-v2/CertificateCard.jsx [hunk 4]
```diff
@@ -37,24 +40,47 @@ const CertificateCard = ({
...
                 defaultMessage="From"
               />
             </p>
             <h5 className="mb-0 color-black">{courseOrganization}</h5>
-            <p className="small mb-0">
+            <p className={classNames([
+              'mb-0',
+              isMobileView ? 'x-small' : 'small',
+            ])}
+            >
```
Comment (sundasnoreen12) [inline src/profile-v2/CertificateCard.jsx:83]: same as above

File: src/profile-v2/CertificateCard.jsx [hunk 5]
```diff
@@ -91,7 +125,8 @@ const CertificateCard = ({
 
 CertificateCard.propTypes = {
   certificateType: PropTypes.string,
-  courseDisplayName: PropTypes.string,
+  courseDisplayName:
+  PropTypes.string,
```
Comment (sundasnoreen12) [inline src/profile-v2/CertificateCard.jsx:129]: move to single line

File: src/profile-v2/Certificates.jsx [hunk 6]
```diff
@@ -4,53 +4,71 @@ import { FormattedMessage } from '@edx/frontend-platform/i18n';
...
+  const isMobileView = useWindowSize().width <= breakpoints.small.minWidth;
+  const isTabletView = useWindowSize().width <= breakpoints.medium.minWidth;
+  return (
+    <div>
+      <div className="col justify-content-start align-items-start g-5rem p-0">
+        <div className="col align-self-stretch height-2625rem justify-content-start align-items-start p-0">
+          <p className={classNames([
+            'font-weight-bold text-primary-500 m-0',
+            isMobileView ? 'h3' : 'h2',
+          ])}
```
Comment (sundasnoreen12) [inline src/profile-v2/Certificates.jsx:22]: follow above pattern for classNames and change wherever used

File: src/profile-v2/Certificates.jsx [hunk 7]
```diff
@@ -4,53 +4,71 @@ import { FormattedMessage } from '@edx/frontend-platform/i18n';
...
-        <p className="font-weight-normal text-gray-800 m-0 p-0">
+      {certificates?.length > 0 ? (
+        <div className="col">
+          <div className={classNames(
+            'row align-items-center pt-5 g-3rem',
+            { 'justify-content-center': isTabletView },
+          )}
+          >
+            {certificates.map(certificate => (
+              <CertificateCard key={certificate.courseId} {...certificate} />
```
Comment (sundasnoreen12) [inline src/profile-v2/Certificates.jsx:56]: You are passing the entire certificate object as a prop. When objects are passed as props, React may treat them as new references even if their contents remain unchanged, potentially causing unnecessary re-renders. To optimize performance, pass only the necessary properties explicitly. This approach helps React better manage re-renders and improves clarity.

File: src/profile-v2/ProfilePage.jsx [hunk 8]
```diff
@@ -7,7 +7,13 @@ import { sendTrackingLogEvent } from '@edx/frontend-platform/analytics';
 import { ensureConfig, getConfig } from '@edx/frontend-platform';
 import { AppContext } from '@edx/frontend-platform/react';
 import { useIntl } from '@edx/frontend-platform/i18n';
-import { Alert, Hyperlink } from '@openedx/paragon';
+import {
+  Alert,
+  Hyperlink,
+  breakpoints,
+  useWindowSize,
+} from '@openedx/paragon';
```
Comment (sundasnoreen12) [inline src/profile-v2/ProfilePage.jsx:15]: import {
  Alert, Hyperlink, breakpoints, useWindowSize,
} from '@openedx/paragon';

File: src/profile-v2/ProfilePage.jsx [hunk 9]
```diff
@@ -42,6 +48,8 @@ const ProfilePage = ({ params }) => {
   } = useSelector(profilePageSelector);
 
   const [viewMyRecordsUrl, setViewMyRecordsUrl] = useState(null);
+  const isMobileView = useWindowSize().width <= breakpoints.small.minWidth;
+  const isTabletView = useWindowSize().width <= breakpoints.medium.minWidth;
```
Comment (sundasnoreen12) [inline src/profile-v2/ProfilePage.jsx:52]: Create generic custom hooks as shown below and reuse them wherever needed:
export function useIsOnTabletScreen() {
  const windowSize = useWindowSize();
  return windowSize.width <= breakpoints.medium.minWidth;
}

export function useIsOnMobileScreen() {
  const windowSize = useWindowSize();
  return windowSize.width <= breakpoints.small.minWidth;
}

File: src/profile-v2/UsernameDescription.jsx [hunk 10]
```diff
@@ -1,18 +1,27 @@
...
-      defaultMessage="Your learner records information is only visible to you. Only your username is visible to others on {siteName}."
-      description="A description of the username field"
-      values={{
-        siteName: getConfig().SITE_NAME,
-      }}
-    />
-  </p>
-);
+const UsernameDescription = () => {
+  const isMobileView = useWindowSize().width <= breakpoints.small.minWidth;
```
Comment (sundasnoreen12) [inline src/profile-v2/UsernameDescription.jsx:8]: reuse created hook

**Review by sundasnoreen12 — CHANGES_REQUESTED** [review]

File: src/profile-v2/CertificateCard.jsx (hunk 2 above)
Comment (sundasnoreen12) [reply src/profile-v2/CertificateCard.jsx:65]: <p className={classNames('m-0 color-black', {
  'h5': isMobileView,
  'h4': !isMobileView,
})}>

File: src/profile-v2/CertificateCard.jsx [hunk 11]
```diff
@@ -27,6 +28,8 @@ const CertificateCard = ({
     audit: null,
   }[certificateType] || null;
 
+  const isMobileView = useWindowSize().width <= breakpoints.small.minWidth;
```
Comment (ayesha-waris) [inline src/profile-v2/CertificateCard.jsx:31]: export function useIsOnMobile() {
  const windowSize = useWindowSize();
  return windowSize.width <= breakpoints.small.minWidth;
}
consider creating hooks for window screens

File: src/profile-v2/Certificates.jsx [hunk 12]
```diff
@@ -4,53 +4,71 @@ import { FormattedMessage } from '@edx/frontend-platform/i18n';
...
-        <h2 className="font-weight-bold text-primary-500 m-0">
-          <FormattedMessage
-            id="profile.your.certificates"
-            defaultMessage="Your certificates"
-            description="heading for the certificates section"
-          />
-        </h2>
+const Certificates = ({ certificates }) => {
+  const isMobileView = useWindowSize().width <= breakpoints.small.minWidth;
+  const isTabletView = useWindowSize().width <= breakpoints.medium.minWidth;
```
Comment (ayesha-waris) [inline src/profile-v2/Certificates.jsx:14]: consider creating hooks for screen sizes as mentioned above

**Review by sundasnoreen12 — APPROVED** [review]

**Pull request merged**